components/publishers/index.jsx

```diff
@@ -0,0 +1,167 @@
+/* Copyright (c) 2015-present, salesforce.com, inc. All rights reserved */
+/* Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import shortid from 'shortid';
+
+import ClassNames from 'classnames';
+
+import { PUBLISHERS } from '../../utilities/constants';
+
+import Avatar from '../avatar';
+import MediaObject from '../media-object';
+import Button from '../button';
+
+// This component's `checkProps` which issues warnings to developers about properties when in development mode (similar to React's built in development tools)
+import checkProps from './check-props';
+import componentDoc from './docs.json';
+
+const propTypes = {
+	/**
+	 * Unique ID for the component to enable ARIA functioning and keyboard interactions
+	 */
+	id: PropTypes.string,
+	/**
+	 * Function callback on click of onAddUser button
+	 */
+	onAddUser: PropTypes.func,
+	/**
+	 * Function callback on click of onAttach button
+	 */
+	onAttach: PropTypes.func,
+	/**
+	 * Function callback on change of textarea content
+	 */
+	onChange: PropTypes.func.isRequired,
+	/**
+	 * Function callback on click of submit button
+	 */
+	onSubmit: PropTypes.func.isRequired,
+	/**
+	 * Placeholder of the text area
+	 */
+	placeholder: PropTypes.string,
+	/**
+	 * This label appears above the text area
+	 */
+	label: PropTypes.string,
```

Comment:
There should be a `labels.comment` for the comment text button, so that it can be changed. Also any "English text" in this component for instance add user `title` should be here. the Share button, etc.